<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/uchida/uchida_ws/src/fusion360files/kataude_description/urdf/kataude.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kataude">
    <material name="silver_default">
        <color rgba="0.700 0.700 0.700 1.000" />
    </material>
    <material name="steel_satin">
        <color rgba="0.6274509803921569 0.6274509803921569 0.6274509803921569 1.0" />
    </material>
    <material name="opaque696969">
        <color rgba="0.27058823529411763 0.27058823529411763 0.27058823529411763 1.0" />
    </material>
    <material name="opaque191191191">
        <color rgba="0.7490196078431373 0.7490196078431373 0.7490196078431373 1.0" />
    </material>
    <transmission name="ShoulderJoint2_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="ShoulderJoint2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="ShoulderJoint2_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="xm540_2_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="xm540_2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="xm540_2_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="xm540_3_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="xm540_3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="xm540_3_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="ShoulderJoint1_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="ShoulderJoint1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="ShoulderJoint1_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="ElbowJoint_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="ElbowJoint">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="ElbowJoint_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="xm540_1_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="xm540_1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="xm540_1_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="EndEffecter1_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="EndEffecter1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="EndEffecter1_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="Endeffecter2_tran">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Endeffecter2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Endeffecter2_actr">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin filename="libgazebo_ros_control.so" name="control" />
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
    <gazebo reference="base_link">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
        <gravity>true</gravity>
    </gazebo>
    <gazebo reference="UpperArm_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="XM540_270R__3_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="mikata_arm_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="Shoulder_Servo3_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="ForeArm_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="XM540_270R_2_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="gripper1_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="gripper2_1">
        <material>Gazebo/Silver</material>
        <mu1>0.2</mu1>
        <mu2>0.2</mu2>
        <selfCollide>true</selfCollide>
    </gazebo>
    <!-- <link name="dummy">
   </link> -->
    <link name="world" />
    <link name="base_link">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.15278443948564172 -1.646071389736825e-05 1.9943676527636152" />
            <mass value="3.145965262261491" />
            <inertia ixx="0.001806" ixy="-0.0" ixz="4e-06" iyy="0.005212" iyz="0.0" izz="0.004982" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/base_link.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque696969" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/base_link.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="UpperArm_1">
        <inertial>
            <origin rpy="0 0 0" xyz="2.3936065328189482e-05 0.06873152009023258 -0.3984141268400523" />
            <mass value="142.51149550389982" />
            <inertia ixx="5.64702" ixy="-0.0" ixz="2.8e-05" iyy="5.488361" iyz="-0.000188" izz="0.635098" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.3025 0.06899 -2.01652" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/UpperArm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="steel_satin" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.3025 0.06899 -2.01652" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/UpperArm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="XM540_270R__3_1">
        <inertial>
            <origin rpy="0 0 0" xyz="0.024533563447407203 -0.00021381174077944674 -0.0631193250864" />
            <mass value="0.09824516357620273" />
            <inertia ixx="6.2e-05" ixy="-0.0" ixz="-2e-06" iyy="5.8e-05" iyz="-0.0" izz="2.8e-05" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.321809 0.000911 -0.457908" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/XM540_270R__3_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque696969" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.321809 0.000911 -0.457908" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/XM540_270R__3_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="mikata_arm_1">
        <inertial>
            <origin rpy="0 0 0" xyz="-9.00741698207086e-05 -0.02460604325871079 -0.04241777280246267" />
            <mass value="0.5291067052461904" />
            <inertia ixx="0.000219" ixy="-1e-06" ixz="-0.0" iyy="0.00018" iyz="-6e-06" izz="0.000224" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.297625 -0.025704 -0.374375" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/mikata_arm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="steel_satin" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.297625 -0.025704 -0.374375" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/mikata_arm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="Shoulder_Servo3_1">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.001267246990176496 0.00020718172021483958 0.029980098166622282" />
            <mass value="4.659806299364297" />
            <inertia ixx="0.006381" ixy="7.7e-05" ixz="-0.006229" iyy="0.074421" iyz="1.2e-05" izz="0.076997" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.2215 2e-05 -1.99849" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/Shoulder_Servo3_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque696969" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.2215 2e-05 -1.99849" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/Shoulder_Servo3_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="ForeArm_1">
        <inertial>
            <origin rpy="0 0 0" xyz="0.00010929466551301559 0.06821266521028407 -0.3727168964631816" />
            <mass value="131.66138695773583" />
            <inertia ixx="4.458919" ixy="0.0" ixz="-0.000184" iyy="4.313539" iyz="-0.00075" izz="0.590679" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.302466 0.069082 -1.250541" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/ForeArm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="steel_satin" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.302466 0.069082 -1.250541" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/ForeArm_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="XM540_270R_2_1">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.0002450048864300891 2.0149603131389355e-05 -0.03209129443533182" />
            <mass value="0.09211846580456283" />
            <inertia ixx="3.9e-05" ixy="0.0" ixz="-0.0" iyy="4.6e-05" iyz="-0.0" izz="2.4e-05" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.297326 0.000884 -0.507317" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/XM540_270R_2_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque696969" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.297326 0.000884 -0.507317" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/XM540_270R_2_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="gripper1_1">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.019092930378784956 -0.008786732838234303 -0.03526207180079688" />
            <mass value="0.3618135799855471" />
            <inertia ixx="0.0003" ixy="0.0" ixz="1e-06" iyy="0.000362" iyz="2.3e-05" izz="0.000133" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.280651 0.018105 -0.328133" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/gripper1_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque191191191" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.280651 0.018105 -0.328133" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/gripper1_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <link name="gripper2_1">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.01899942595107912 0.009210263060438971 -0.027914083411589796" />
            <mass value="0.27970437781382335" />
            <inertia ixx="0.000199" ixy="-0.0" ixz="-0.0" iyy="0.000246" iyz="-8e-06" izz="0.000104" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0.280651 -0.013895 -0.328133" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/gripper2_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
            <material name="opaque191191191" />
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0.280651 -0.013895 -0.328133" />
            <geometry>
                <mesh filename="package://kataude_description/meshes/gripper2_1.stl" scale="0.001 0.001 0.001" />
            </geometry>
        </collision>
    </link>
    <!-- <joint name="dummy_joint" type="fixed">
     <parent link="dummy"/>
     <child link="base_link"/>
   </joint> -->
    <joint name="world_joint" type="fixed">
        <parent link="world" />
        <child link="base_link" />
        <origin rpy="0 0 0" xyz="0 0 0" />
    </joint>
    <joint name="ShoulderJoint1" type="revolute">
        <origin rpy="0 0 0" xyz="-0.2215 -2e-05 1.99849" />
        <parent link="base_link" />
        <child link="Shoulder_Servo3_1" />
        <axis xyz="1.0 -0.0 -0.0" />
        <limit effort="100" lower="-1.570796" upper="1.570796" velocity="100" />
    </joint>
    <joint name="ShoulderJoint2" type="revolute">
        <origin rpy="0 0 0" xyz="-0.081 -0.06897 0.01803" />
        <parent link="Shoulder_Servo3_1" />
        <child link="UpperArm_1" />
        <axis xyz="0.0 1.0 0.0" />
        <limit effort="100" lower="0.0" upper="3.141593" velocity="100" />
    </joint>
    <joint name="ElbowJoint" type="revolute">
        <origin rpy="0 0 0" xyz="3.4e-05 -9.2e-05 -0.765979" />
        <parent link="UpperArm_1" />
        <child link="ForeArm_1" />
        <axis xyz="-0.0 1.0 0.0" />
        <limit effort="100" lower="-1.570796" upper="1.570796" velocity="100" />
    </joint>
    <joint name="xm540_1" type="continuous">
        <origin rpy="0 0 0" xyz="0.00514 0.068198 -0.743224" />
        <parent link="ForeArm_1" />
        <child link="XM540_270R_2_1" />
        <axis xyz="0.0 -0.0 -1.0" />
    </joint>
    <joint name="xm540_2" type="revolute">
        <origin rpy="0 0 0" xyz="-0.024483 -2.7e-05 -0.049409" />
        <parent link="XM540_270R_2_1" />
        <child link="XM540_270R__3_1" />
        <axis xyz="1.0 0.0 0.0" />
        <limit effort="100" lower="-1.570796" upper="1.570796" velocity="100" />
    </joint>
    <joint name="xm540_3" type="revolute">
        <origin rpy="0 0 0" xyz="0.024184 0.026615 -0.083533" />
        <parent link="XM540_270R__3_1" />
        <child link="mikata_arm_1" />
        <axis xyz="-0.0 1.0 -0.0" />
        <limit effort="100" lower="-1.570796" upper="1.570796" velocity="100" />
    </joint>
    <joint name="EndEffecter1" type="revolute">
        <origin rpy="0 0 0" xyz="0.016974 -0.043809 -0.046242" />
        <parent link="mikata_arm_1" />
        <child link="gripper1_1" />
        <axis xyz="1.0 0.0 0.0" />
        <limit effort="100" lower="-1.047198" upper="0.0" velocity="100" />
    </joint>
    <joint name="Endeffecter2" type="revolute">
        <origin rpy="0 0 0" xyz="0.016974 -0.011809 -0.046242" />
        <parent link="mikata_arm_1" />
        <child link="gripper2_1" />
        <axis xyz="1.0 0.0 0.0" />
        <limit effort="100" lower="0.0" upper="1.047198" velocity="100" />
    </joint>
</robot>
